src/core/net/ip6.cpp

```diff
@@ -1054,6 +1054,17 @@ Error Ip6::ProcessReceiveCallback(Message &          aMessage,
         }
     }
 
+    if (aIpProto == kProtoFragment && mIcmp.ShouldHandleEchoRequest(aMessageInfo))
```

Comment:
I feel that the logic should be in `if (mIsReceiveIp6FilterEnabled && aAllowReceiveFilter)` block. 

Thoughts? @abtink @lmaciejonczyk

Reply:
It cannot be there because _aAllowReceiveFilter_ is always _false_ for IPv6 fragments in the current implementation.